src/thx/fp/List.hx

```diff
@@ -16,7 +16,15 @@ abstract List<A>(ListImpl<A>) from ListImpl<A> to ListImpl<A> {
 
   @:from
   inline static public function fromArray<A>(arr : Array<A>) : List<A>
-    return arr.reduceRight.fn(bin(_1, _0), empty());
+    return arr.reduceRight(
+      function(memo,next){
+        return switch(memo){
```

Comment:
This switch is unnecessary. You could just write `arr.reduceRight(function(memo, next) return Cons(next, memo)`.